### PR TITLE
Add client-hints clearing support

### PR DESCRIPTION
**Pull request by arichiv**
We should be sure to cover this too for a new specific option, as well as the existing cookie and cache cases.

https://github.com/WICG/client-hints-infrastructure/issues/155

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-clear-site-data/pull/74.html" title="Last updated on Jul 14, 2023, 12:04 PM UTC (9f71037)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-clear-site-data/74/5dbb850...9f71037.html" title="Last updated on Jul 14, 2023, 12:04 PM UTC (9f71037)">Diff</a>

### CODE REVIEW

**Review by mikewest — COMMENTED** [review]
Definition looks reasonable enough. I'm curious about why we'd separate this kind of cache out from `cache` or `storage`? The other options we currently have are fairly broad (`storage` vs. `indexed-db`/`localstorage`/etc); do you have thoughts about a guideline we could follow that would help us land on the right level of granularity?

**Comment by arichiv** [comment]
That’s a good point, I’m not sure I have a hard and fast line to adhere to. This situation feels unique since the only way to clear the CH cache is via a header already, but the existing method (an empty Accept-CH header) was against the spec for an sf-list and would fail if a second header of the same type was sent. There likely may be a reason to clear the cache or cookies at the same time that the ch cache is cleared, but because we already had a targeted way to clear it the granularity felt reasonable.

Another option would be to just clear the ch cache on cache or cookie clears, and not add a granular target. Instead someone could send `Accept-CH: clear`. This would be compliant with the sf-list spec and even if other headers of the same type are sent we could just roll up the values, notice the clear token, and ignore all but it. @miketaylr @yoavweiss for comment

File: index.src.html [hunk 1]
```diff
@@ -306,6 +307,10 @@ <h3 id="header">
         </pre>
       </div>
 
+      Note: Clearing cookies should also clear the [=Accept-CH cache=] for
+      <a>origin</a>. This is because the cache is also cleared if the user
+      manually clears cookies.
```
Comment (mikewest) [inline index.src.html:312]: Perhaps you could also update CH infra to include this requirement? Around https://wicg.github.io/client-hints-infrastructure/#accept-ch-cache-definition:~:text=As%20such%2C%20a%20user%20agent%20MUST%20evict%20that%20cache%20whenever%20the%20user%20clears%20their%20cookies%20or%20when%20session%20cookies%20expire. looks right.

File: index.src.html [hunk 2]
```diff
@@ -471,6 +496,8 @@ <h3 id="clear-response" algorithm>
             ::  <a lt="clear cookies" abstract-op>Clear cookies for |origin|</a>.
             :   "<a grammar>`storage`</a>"
             ::  <a lt="clear storage" abstract-op>Clear DOM-accessible storage for |origin|</a>.
+            :   "<a grammar>`clientHints`</a>"
+            ::  [=map/Set=] [=Accept-CH cache=][|origin|] to an empty [=ordered set=].
```
Comment (mikewest) [inline index.src.html:500]: Sets are lists, so you could use https://infra.spec.whatwg.org/#list-empty for a slightly more pithy instruction. That is:

```suggestion
            ::  [=list/Empty=] [=Accept-CH cache=][|origin|].
```

**Review by mikewest — APPROVED** [review]
LGTM % nits.

File: index.src.html (hunk 1 above)
Comment (arichiv) [reply index.src.html:312]: I will, was waiting on this first

**Pull request merged**